Feature/server update api

talorlangnas-moveo opened the pull request.

talorlangnas-moveo committed:
feat: enhance restaurant API with new filter functions, query parameters and moment.js integration

- Added query parameters to the findAll method in RestaurantsController for filtering by founded date, rating, and open status.
- Implemented new service methods to retrieve the top 3 newest and most popular restaurants, as well as to find currently open restaurants using moment.js for time handling.
- Included moment.js as a dependency in package.json and updated pnpm-lock.yaml.

talorlangnas-moveo committed:
feat: enhance chefs API with new query parameters for filtering

- Added query parameters to the findAll method in ChefsController for filtering by founded date and number of views.
- Implemented new service methods to retrieve the top 3 newest and most popular chefs.

talorlangnas-moveo committed:
refactor: update DataDisplay component and restaurant filtering logic

- Removed generic type from DataDisplay component for improved clarity.
- Updated filter functions to return promises, enhancing asynchronous handling.
- Consolidated restaurant filtering options and removed redundant utility functions.
- Added new API methods for fetching the newest, most popular, and currently open restaurants as cards.

talorlangnas-moveo committed:
refactor: streamline chefs and restaurants filtering logic and update styles

- Replaced utility imports with service imports for chefs and restaurants filtering options.
- Removed obsolete filter functions to simplify the codebase.
- Updated DataDisplay component to enhance clarity by removing generic types.
- Adjusted styles in dataDisplay module for improved layout consistency.

talorlangnas-moveo committed:
initializing next js project

talorlangnas-moveo committed:
chore: update package dependencies and enhance TypeScript configuration

- Updated package.json to include new dependencies and maintain existing ones.
- Added path mapping for styles in tsconfig.json to improve module resolution.
- Commented out unused imports and code in layout.tsx for cleaner structure and future reference.
- Imported main.scss for styling integration.

talorlangnas-moveo committed:
initialized epicure-admin-app nextjs project

talorlangnas-moveo committed:
creating restaurants route

talorlangnas-moveo committed:
creating data-table component

talorlangnas-moveo committed:
feat: create Restaurants pages with data table integration

- Created `dataDisplay` component to display data generically
- Implemented a `data-table` using shadcn UI components
- Defined `Restaurant` interface to structure restaurant data
- Created `RestaurantColumn` type for table column definitions

talorlangnas-moveo committed:
adding dishes images and delete payments directory

talorlangnas-moveo committed:
creating adim dishes page

talorlangnas-moveo committed:
creating basic files

talorlangnas-moveo committed:
creating admin chefs page

talorlangnas-moveo committed:
refactor: update dish data structure and improve type definitions

- Renamed `title` to `name` in the Dish interface for consistency.
- Introduced `DishCategoryType` to enforce specific dish categories.
- Updated related components and services to reflect the new naming and type changes.
- Cleaned up unused imports in the dishes column file.

talorlangnas-moveo committed:
Merge branch 'feature/admin-dishes-page' into feature/admin-chefs-page

talorlangnas-moveo committed:
refactor: rename image property to imgUrl in dish data structure

- Updated the property name from `image` to `imgUrl` for consistency across the dish data structure.
- Ensured all instances of the property were updated accordingly in the dish data retrieval function.

talorlangnas-moveo committed:
refactor: update accessor key from image to imgUrl in dishes column definition  - Changed the accessor key from `image` to `imgUrl` in the dishes column configuration for consistency with the updated data structure.

talorlangnas-moveo committed:
chore: update dependencies and enhance sidebar styles

- Updated global CSS to define sidebar color variables for light and dark themes.
- Refactored layout to include a sidebar provider and trigger for improved UI structure.

talorlangnas-moveo committed:
refactor: adjust padding in DataDisplay component

- Updated padding in the DataDisplay component for improved layout consistency.

talorlangnas-moveo committed:
feat: enhance Home and Restaurants pages with new components and data structure

- Introduced a new `getItems` function in the Home component to fetch card data for the admin dashboard.
- Updated the Home component to display a `Features` section with dynamic content.
- Modified the Restaurants page to include a `StatsSection` for better data presentation within the `DataDisplay` component.
- Adjusted the `DataDisplay` component to make the title optional and improve layout flexibility.

talorlangnas-moveo committed:
refactor: enhance DataDisplay component and update page structures

- Added an optional `description` prop to the DataDisplay component for improved content presentation.
- Updated the Home component to include a blank line for better readability.
- Refactored the Restaurants page to streamline the DataDisplay usage by directly passing the title prop.

talorlangnas-moveo committed:
feat: implement CRUD operations for restaurants

- Added update and delete methods in the RestaurantsController and RestaurantsService.
- Updated existing methods to return appropriate types for better type safety.
- Enhanced the controller with new routes for updating and removing restaurants.

talorlangnas-moveo committed:
feat: implement CRUD operations for dishes

- Added update and delete methods in the DishesController and DishesService.
- Updated existing methods to return appropriate types for better type safety.
- Enhanced the controller with new routes for updating and removing dishes.

talorlangnas-moveo committed:
feat: implement CRUD operations for chefs

- Added update and delete methods in the ChefsController and ChefsService.
- Updated existing methods to return appropriate types for better type safety.
- Enhanced the controller with new routes for updating and removing chefs.

taluliel18 commented on epicure-server/src/modules/restaurants/restaurants.service.ts:
```diff
@@ -31,4 +33,58 @@ export class RestaurantsService {
+    return this.restaurantModel.find().sort({ rating: -1 }).limit(3).exec();
+  }
+
+  async getOpenRestaurantsNow(): Promise<Restaurant[]> {
+    const now = moment();
+
+    const allRestaurants = await this.restaurantModel.find().exec();
+
+    const openRestaurants = allRestaurants.filter((restaurant) => {
+      const opening = moment(restaurant.openingTime, 'HH:mm');
+      const closing = moment(restaurant.closingTime, 'HH:mm');
+
+      if (closing.isBefore(opening)) {
+        return now.isAfter(opening) || now.isBefore(closing);
+      } else {
+        return now.isBetween(opening, closing);
+      }
+    });
+
+    return openRestaurants;
+  }
```
Consider replacing the in-memory filtering logic in getOpenRestaurantsNow() with a MongoDB aggregation pipeline. This will reduce the need to load all restaurant documents into memory and improve performance, especially as the dataset grows.

talorlangnas-moveo replied:
done

taluliel18 commented on epicure-server/src/modules/restaurants/restaurants.service.ts:
```diff
@@ -31,4 +33,58 @@ export class RestaurantsService {
     return restaurant;
   }
+
+  async updateById(
+    id: Types.ObjectId,
+    updateRestaurantDto: UpdateRestaurantDto,
+  ): Promise<Restaurant> {
+    const updatedRestaurant = await this.restaurantModel.findByIdAndUpdate(
+      id,
+      updateRestaurantDto,
+      { new: true },
+    );
+    if (!updatedRestaurant) {
+      throw new NotFoundException('Restaurant not found');
+    }
+    return updatedRestaurant;
+  }
+
+  async removeById(id: Types.ObjectId): Promise<Restaurant> {
+    const deletedRestaurant = await this.restaurantModel.findByIdAndDelete(id);
+    if (!deletedRestaurant) {
+      throw new NotFoundException('Restaurant not found');
+    }
+    return deletedRestaurant;
+  }
```
Consider simplifying method names in RestaurantService to follow standard CRUD naming conventions (e.g., update, delete, findById). Since the service name already implies the domain (restaurants), adding "ById" or "Restaurant" in method names might be redundant.

talorlangnas-moveo replied:
done

taluliel18 commented on epicure-admin-app/src/components/features-1.tsx:
```diff
@@ -0,0 +1,71 @@
+  cards?: CardItem[];
+}
+
+export default function Features({
```
What is this component? What does "features" mean? And why is the file name different from the component name?

talorlangnas-moveo replied:
done

talorlangnas-moveo committed:
feat: optimize getOpenRestaurantsNow method for better performance

- Replaced the previous filtering logic with an aggregation pipeline to determine open restaurants.
- Added handling for overnight hours and current time comparison directly in the database query.
- Improved code readability and efficiency by utilizing MongoDB's aggregation framework.

talorlangnas-moveo committed:
refactor: standardize method names in chefs, dishes, and restaurants modules

- Renamed methods in ChefsController, DishesController, and RestaurantsController from `findById`, `updateById`, and `removeById` to `findOne`, `update`, and `remove` for consistency across modules.
- Updated corresponding service methods in ChefsService, DishesService, and RestaurantsService to reflect the new naming conventions.

talorlangnas-moveo committed:
feat: replace Features component with StatsPanel in Home page

- Updated the Home component to utilize the new StatsPanel component instead of the deprecated Features component.
- Removed the Features component file to streamline the codebase.

talorlangnas-moveo committed:
Merge branch 'development' into feature/server-update-api
